### PR TITLE
Implementation a Vert.x backend for Tapir

### PR DESCRIPTION
**Pull request by aesteve**
Very first working implementation:
* [x] tests are passing
* [x] still a few todos / fixmes to address
* [x] lots of stuff need to be configurable through server options
* [x] needs a huge refactoring
* [x] needs documentation
* [x] respect standards (scalafmt, etc.)

### CODE REVIEW

**Comment by aesteve** [comment]
This was meant to be a PR on my own fork. Sorry about the noise, it's totally not ready!

**Comment by adamw** [comment]
No worries :D

**Comment by adamw** [comment]
Compilation seems to fail

File: server/vertx/src/main/scala/sttp/tapir/server/vertx/decoders/VertxInputDecoders.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        val decodeFailureCtx = DecodeFailureContext(input, failure)
+        endpointOptions.decodeFailureHandler(decodeFailureCtx) match {
+          case DecodeFailureHandling.NoMatch =>
+            endpointOptions.logRequestHandling.decodeFailureNotHandled(endpoint, decodeFailureCtx)(endpointOptions.logger)
+            rc.response.setStatusCode(404).end()
+          case DecodeFailureHandling.RespondWithResponse(output, value) =>
+            endpointOptions.logRequestHandling.decodeFailureHandled(endpoint, decodeFailureCtx, value)(endpointOptions.logger)
+            VertxOutputEncoders.apply(output, value)(endpointOptions)(rc)
+        }
+    }(endpointOptions.executionContextOr(VertxExecutionContext(rc.vertx.getOrCreateContext)))
```
Comment (adamw) [inline server/vertx/src/main/scala/sttp/tapir/server/vertx/decoders/VertxInputDecoders.scala:56]: This seems to occur in a couple of places - maybe this EC-providing method could be extracted somewhere?>

**Comment by adamw** [comment]
See https://github.com/aesteve/tapir/pull/2 - fixes the compilation :)

**Comment by aesteve** [comment]
It compiles, and tests are running on my computer though.
What's the compilation issue?

**Comment by adamw** [comment]
See travis :) I suspect you have java 11+, and the build runs with java8

**Comment by aesteve** [comment]
I merged your PR a bit too early. I guess there's a bug in the buffer conversion.
You seem to be always appending to the start of the buffer.

**Comment by adamw** [comment]
Ah, sttp-model was recently bumped to 2.1, and there `MultiQueryParams` is replaced with `QueryParams`. You probably will need to refresh the project in the IDE to see that :)

**Comment by adamw** [comment]
Let me know when you'll think that this will be ready to merge :)

**Comment by aesteve** [comment]
Sure, thanks.

I'll review everything tomomrrow or the day after, to get a fresh view.

**Comment by aesteve** [comment]
I got rid of every "blocking" implementations.

Will give it a final look this week-end in case I forgot about something and it should be ok.

**Comment by adamw** [comment]
Great!

**Comment by aesteve** [comment]
I haven't forgotten, just ran a bit out of time.
It's probably mergeable, if you're in a hurry. 
If you're not, I'll try to give it a better look this week-end.

**Comment by adamw** [comment]
@aesteve No worries, we'll merge when you're happy with the PR :)

**Comment by aesteve** [comment]
I guess it should be mergeable.

**Comment by adamw** [comment]
Released in 0.15.0. Thanks!

**Pull request merged**